activestorage/lib/active_storage/previewer/poppler_pdf_previewer.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+module ActiveStorage
+  class Previewer::PopplerPDFPreviewer < Previewer
+    def self.accept?(blob)
+      blob.content_type == "application/pdf" &&
+        system("#{pdftoppm_path} -v")
```

Comment:
Also memoize this.

Reply:
This should use the multi-argument form, in case the path contains a space